doc/implementation_comparison.md

```diff
@@ -0,0 +1,198 @@
+# grpc-dotnet and Grpc.Core comparison
+
+This document summarizes the differences between the two available implementations of gRPC in C#.
+
+## Where both implementations are the same
+
+- 100% wire compatible and interoperable with each other and with other gRPC implementations
+- same API for invoking and handling RPC calls
+- basic functionality (streaming, metadata, deadline, cancellation, ...)
+- using same codegen tooling and MSBuild Integration (Grpc.Tools)
+- differences in setting up server and client are documented here: https://docs.microsoft.com/en-us/aspnet/core/grpc/migration?view=aspnetcore-3.0
+
+## Criteria for choosing between grpc-dotnet and gRPC C#
+
+One might choose one or the other implementation mostly for one of these reasons
+
+- avoid use of native code
+- ability to use .NET Core 3 and ASP.NET Core 3 (it's a brand new stack so not everyone will be able to use immediately)
+- want seamless integration with ASP.NET Core 3, dependency injection etc.
+- features available (see breakdown)
+- maturity level
+- performance (TODO: add data)
+
+## Comparison of supported features 
+
+Beyond the basic RPC functionality, there are a lot of gRPC features that may or may not be supported. The summary
+of supported features in both implementation is available in this section.
+
+### Service config
+
+Service config is currently only supported by Grpc.Core. Right now the feature is not that useful 
+because support for service config encoded in DNS records hasn't been enabled yet by default.
+
+Also see:
+
+- https://github.com/grpc/grpc/blob/master/doc/service_config.md
+- https://github.com/grpc/proposal/blob/master/A2-service-configs-in-dns.md 
+
+### Tracing
+
+grpc-dotnet has tracing bindings that come from ASP.NET Core, and it works with OpenTelemetry SDK, OpenCensus support is TBD.
```

Comment:
Let's link to the sample added by https://github.com/grpc/grpc-dotnet/pull/570 when it goes in.